### ISSUE TITLE
problem with returned Optional

### ISSUE THREAD

**Issue by ikod** [aliak00/optional#26]
Hello,

Is it expected that this code throws assertion, or I'm doing something wrong?

```
/+dub.sdl:
dependency "optional" version="~>0.7.3"
+/
import optional;

Optional!string f()
{
    Optional!string v;
    return v;
}

void main()
{
    Optional!string o;
    o = f();
    assert(o.empty, "must be empty");
}

```

**Comment by aliak00** [comment]
Fix released in 0.7.4

**Comment by ikod** [comment]
Thanks!

пт, 16 нояб. 2018 г., 13:07 Ali Akhtarzada notifications@github.com:
